### PR TITLE
Fix the problem of multiple namespaces in MetadataReport

### PR DESCRIPTION
**Pull request by finefuture**
## What is the purpose of the change

Key missing namespace information:
image

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@chickenlj PTAL

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReportFactory.java [hunk 1]
```diff
@@ -89,6 +91,12 @@ public MetadataReport getMetadataReport(URL url) {
     }
 
     protected String toMetadataReportKey(URL url) {
+        String namespace = url.getParameter(NAMESPACE_KEY);
+        if (!StringUtils.isEmpty(namespace)) {
+            return URL.valueOf(url.toServiceString())
+                    .addParameter(NAMESPACE_KEY, namespace)
+                    .toString();
+        }
```
Comment (chickenlj) [inline dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReportFactory.java]: Can you describe the difference between the new url and the old url?

Comment (chickenlj) [reply dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReportFactory.java]: How about making it a public util method to make calling it in other places easier? like here https://github.com/apache/dubbo/pull/13971/files#diff-17a0f234d88728ebc6f37959d84276704ca16846cd10cd0743616a8689949eabR33

**Comment by chickenlj** [comment]
I think it would be better to just change the way how `REGISTRY_CLUSTER_KEY` is generated.

For now, `REGISTRY_CLUSTER_KEY` is the same as `id` (controlled by `@Parameter(key = REGISTRY_CLUSTER_KEY)`). If we want it to include `namespace` in `REGISTRY_CLUSTER_KEY`, we can change the code to make sure that `namespace` is included in `REGISTRY_CLUSTER_KEY` the first time the `registry url` is generated. 

Then the only place we need to change except for the `REGISTRY_CLUSTER_KEY` generating process (to include both id and namespace) is how to map the MetadataReport instance with the  `REGISTRY_CLUSTER_KEY`(also to have `namespace` included) https://github.com/apache/dubbo/pull/13971/files#diff-1e44977c9c4b360c444eb20fbc5599acc73533c783c897954e068e88ab339278R117.

**Comment by finefuture** [comment]
> I think it would be better to just change the way how `REGISTRY_CLUSTER_KEY` is generated.
> 
> For now, `REGISTRY_CLUSTER_KEY` is the same as `id` (controlled by `@Parameter(key = REGISTRY_CLUSTER_KEY)`). If we want it to include `namespace` in `REGISTRY_CLUSTER_KEY`, we can change the code to make sure that `namespace` is included in `REGISTRY_CLUSTER_KEY` the first time the `registry url` is generated.
> 
> Then the only place we need to change except for the `REGISTRY_CLUSTER_KEY` generating process (to include both id and namespace) is how to map the MetadataReport instance with the `REGISTRY_CLUSTER_KEY`(also to have `namespace` included) https://github.com/apache/dubbo/pull/13971/files#diff-1e44977c9c4b360c444eb20fbc5599acc73533c783c897954e068e88ab339278R117.

Ok, I will try it.

**Comment by finefuture** [comment]
@chickenlj PTAL

File: dubbo-common/src/main/java/org/apache/dubbo/config/RegistryConfig.java [hunk 2]
```diff
@@ -220,12 +219,6 @@ public RegistryConfig(ApplicationModel applicationModel, String address, String
         setProtocol(protocol);
     }
 
-    @Override
-    @Parameter(key = REGISTRY_CLUSTER_KEY)
-    public String getId() {
-        return super.getId();
-    }
-
```
Comment (chickenlj) [inline dubbo-common/src/main/java/org/apache/dubbo/config/RegistryConfig.java]: Please keep this method and only remove `@Parameter(key = REGISTRY_CLUSTER_KEY)`.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/config/RegistryConfig.java (hunk 2 above)
Comment (finefuture) [reply dubbo-common/src/main/java/org/apache/dubbo/config/RegistryConfig.java]: Fixed.

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReportFactory.java (hunk 1 above)
Comment (finefuture) [reply dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReportFactory.java]: Add a new method for extra parameters.
image

**Comment by finefuture** [comment]
@chickenlj Does DynamicConfiguration need to support multiple namespaces?
image

**Comment by chickenlj** [comment]
Sorry for the late response, I am working on this pull request now.

**Review by chickenlj — APPROVED** [review]
LGTM.

**Comment by chickenlj** [comment]
@finefuture thanks for the patience and contribution, we do appreciate it.

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**